admin_panel/src/commonMain/kotlin/net/thechance/mena/admin_panel/presentation/screen/deposit/DepositViewModel.kt

```diff
@@ -0,0 +1,163 @@
+import net.thechance.mena.admin_panel.presentation.model.SnackBarState
+import net.thechance.mena.admin_panel.presentation.utils.StringProvider
+import net.thechance.mena.admin_panel.presentation.utils.formatAmount
+import net.thechance.mena.admin_panel.presentation.utils.getErrorSnackBarMsg
+import net.thechance.mena.admin_panel.presentation.utils.getErrorSnackBarTitle
+import net.thechance.mena.admin_panel.resources.Res
+import net.thechance.mena.admin_panel.resources.success_deposit_description
+import net.thechance.mena.admin_panel.resources.success_deposit_title
+
+import org.koin.android.annotation.KoinViewModel
+import org.koin.core.annotation.Provided
+
+@KoinViewModel
+class DepositViewModel (
+    @Provided
+    private val depositMoneyUseCase : DepositMoneyUseCase,
+    @Provided
+    private val stringProvider: StringProvider,
+    @Provided
+    private val depositMoneyRepository: DepositMoneyRepository,
+    private val dispatcher: CoroutineDispatcher = Dispatchers.IO,
+): BaseViewModel<DepositScreenState, Unit>(DepositScreenState()), DepositInteractionListener
+{
+    init {
+        getAvailableCountries()
+    }
+
+    override fun mapError(throwable: Throwable): ErrorState {
+        return when (throwable) {
+            is NoInternetException -> ErrorState.NoInternet
+            is InvalidPhoneNumberException-> DepositErrorState.NoAccount
+            else -> ErrorState.UnknownError
+        }
+    }
+
+    override fun onFillTheWalletButtonClicked() {
+        tryToExecute(
+            onStart = { updateState { it.copy(isDepositProcessLoading = true) } },
+            callee=::onFillWalletClicked,
+            onSuccess = { onDepositSuccess() },
+            onError = ::onDepositError,
+            dispatcher = dispatcher
+        )
+    }
+
+
+    override fun onPhoneNumberChanged(phoneNumber : String) {
+        updateState { it.copy(phoneNumber = phoneNumber) }
+    }
+
+    override fun onAmountChanged(amount : String) {
+        updateState { it.copy(amount = formatAmount( amount)) }
+    }
+
+    override fun onCountryCodeChanged(country: DepositScreenState.CountryUiState) {
+        updateState { it.copy(selectedCountry = country) }
+    }
+
+    private suspend fun onFillWalletClicked(){
+        depositMoneyUseCase.deposit(
+            phoneNumber = currentState.phoneNumber,
+            amount = currentState.amount.replace(",", "").toDouble(),
+            currentState.selectedCountry.toEntity()
+        )
+    }
+
+    private suspend fun onDepositSuccess(){
+        updateState { it.copy(isDepositProcessLoading = false) }
+        showSnackBar(
+            title = stringProvider.getString(Res.string.success_deposit_title),
+            message = stringProvider.getString(Res.string.success_deposit_description),
+            isSuccess = true
+        )
+        updateState {
+            it.copy(
+                phoneNumber = "",
+                amount =""
+            )
+        }
+
+    }
```

Comment:
I think this function would be more readable this way:

```suggestion
    private suspend fun onDepositSuccess(){
        updateState { it.copy(isDepositProcessLoading = false) }
        
        showSnackBar(
            title = stringProvider.getString(Res.string.success_deposit_title),
            message = stringProvider.getString(Res.string.success_deposit_description),
            isSuccess = true
        )
        
        updateState { it.copy(phoneNumber = "", amount ="") }
    }
```